### ISSUE TITLE
Distance Calculator only works once in Circle tab in ArcMap

### ISSUE THREAD

**Issue by Dbarnes1** [Esri/distance-direction-addin-dotnet#194]
I'm struggling to create circles using the Distance Calculator in Distance and Direction. I can create it once by selecting a center point, and then expanding the Distance Calculator and entering a time and speed. However, when I clear graphics and try to create a circle using time and speed again, I run into issues. I can enter time and speed in the distance calculator, but when I click enter a graphic doesn't draw.  

To repro:
Open distance and direction add-in in ArcMap (I'm using 10.5 build 6488)
Navigate to the circle tab
Click a center point using the map point tool
expand the distance calculator
Enter parameters for Time and Speed and click enter. You should see a correct graphic.

Now, clear the graphic and try and create another circle using the distance calculator. I can't get it to work, regardless of whether I close and open the distance calculator section, click a new center point and clear graphics, or change the numbers in the time and minutes boxes. 

If we are short on time we could always push the Distance Calculator functionality back from the version-next release @nfeuerstein @lfunkhouser

**Comment by pHill5136** [comment]
@Dbarnes1 @nfeuerstein Noticed that repopulating both time and rate makes tool respond. Both fields must be refreshed in no particular order. can this issue be addressed through doc or should we pursue a fix?

**Comment by Dbarnes1** [comment]
@pHill5136 I can get it to work now. A refresh fields button would be really useful for when users aren't properly refreshing both fields. Also, what if I try to create the same size circle multiple times from multiple starting locations. It seems like that bug still exists.

**Comment by Thargoids** [comment]
The problem appears to be: if values are present in Time and Rate fields before a point is added, the Radius / Diameter field does not update when the point is added. This then causes a problem in code because if the Distance variable (as obtained from Radius / Diameter) is zero, no graphic is drawn.

The problem does not occur on very first usage of the tool, so it seems like the solution could be to ensure the tool gets reset to this initial state after each creation of a graphic.

I will investigate a fix.

**Comment by Thargoids** [comment]
Looking at this again, I think there is an inconsistency between the Radius / Diameter field being cleared after adding the graphic, whereas the Time / Rate fields remain populated. Surely all three of Radius / Diameter, Time and Rate fields should be cleared, or if Time and Rate are to remain populated then so should Radius / Diameter, because presence of values in Time and Rate provides us with the inputs we need to generate Radius / Diameter.

I am clarifying this with @kgozango.

**Comment by Thargoids** [comment]
@kgonzago informs me that following creation of the circle, the Time and Rate fields should be cleared along with the Distance field

**Comment by nfeuerstein** [comment]
@kgonzago @Thargoids can this be moved to verify?

**Comment by mfunk** [comment]
No. They rolled back the changes from the original. Looks like it's still in progress.

**Comment by Thargoids** [comment]
A fix is ready but I made a mistake with the pull request, which necessitated the roll back. I will create another PR.

The change I have made sets Time and Rate to zero after each point is created. This resolves the issue whereby the fields had to be refreshed after creating a point, but note it means that if a user wants to create multiple circles with the same Time/Rate, they have to reenter these values each time. I've gone with @kgonzago's direction here.
